### PR TITLE
DTNN-132 Clean up bplib prototype readme

### PR DESCRIPTION
**Pull request by gskenned**
**Describe the contribution**
Updates the main README with descriptive text and example configure/build/test scripts.

**Testing performed**
Tested all steps described in:
  - The main README

**Expected behavior changes**
The README update will benefit users with tested, reusable configure/build/test scripts.

**System(s) tested on**
 - Hardware: AWS Elastic Container
 - OS: Ubuntu 20.04.6 LTS

**Contributor Info - All information REQUIRED for consideration of pull request**
Grafton Kennedy, Vantage Systems, Inc. (grafton.s.kennedy@nasa.gov)

### CODE REVIEW

**Comment by czogby-nasa** [comment]
/doc/example-scripts/stand-alone/README.md:

"Copy the example scripts to $BPLIB_HOME."
This isn't specific enough.  It should say "Copy the example scripts in this directory to $BPLIB_HOME".

**Comment by czogby-nasa** [comment]
/doc/example-scripts/stand-alone/README.md:

"Verify that BPLIB_SOURCE in bplib-env-vars is the path to a clone of the bplib repository."

Isn't this script supposed to be installing the bplib repository from scratch?  If so, this line is impossible.

**Comment by czogby-nasa** [comment]
/doc/example-scripts/README.md:

"Verify that environment variables in $CFS_HOME/cfs-env-vars are set to the correct paths."
This should note that most of the paths shouldn't exist yet.  As it is, it makes the reader think the paths should already exist and cFS should already be cloned.

**Comment by gskenned** [comment]
Commit [49f6450](https://github.com/gskenned/bplib/commit/49f645039307454a6bbffc0fd558b36bc92a3dd1) fixes:

Comment: https://github.com/nasa/bplib/pull/260#issuecomment-2239933948

> The /doc/example-scripts/bplib-unit-test-functional sometimes encounters a CMake error:

**Resolution**
Appended a line to set the NasaOsal_DIR in /doc/example-scripts/setup-osal.
NasaOsal_DIR provides the path for bplib-unit-test-functional `CMake` to find the right file.

The error happened when bplib-unit-test-functional ran on a system that did not have OSAL installed. This was a good catch, because I didn't catch it in development.

**Tests Performed**
1. Reproduce the error environment by "uninstalling" existing OSAL installations in the test environment. In my case, I had to remove the OSAL installation done by bp/docs/example-scripts/build-bp `make DESTDIR=$CFS_HOME/cfs-build-folder -j2 mission-install`.
2. Run the steps in bplib/doc/example-tests/README to demonstrate the fix works.

**Comment by gskenned** [comment]
Commit (06749141) fixes:

Comments:
https://github.com/nasa/bplib/pull/260#issuecomment-2237684478
https://github.com/nasa/bplib/pull/260#issuecomment-2237733555
https://github.com/nasa/bplib/pull/260#issuecomment-2237733555

**Resolution**
Revised the bplib/doc/example-scripts/README.md and bplib/doc/example-scripts/stand-alone/README.md as indicated in the comments.

File: README.md [hunk 1]
```diff
@@ -39,99 +39,120 @@ library.
 
 #### Prerequisites
 
-1. The build requires the __cmake__ build system and a compiler toolchain (by default __gcc__).
+1. The build requires Ubuntu 22.04.4 LTS, the __cmake__ build system and a compiler toolchain (by default __gcc__).
```
Comment (dzbaker) [inline README.md:42]: Recommend something along the lines of "BPLib has been tested on Ubuntu 22.04.4 LTS, although it is expected to work on other POSIX-compliant operating systems."

**Comment by gskenned** [comment]
Updated, reviewed, tested, and ready to go.

**Pull request merged**